### PR TITLE
Expose left/right panning as a pref

### LINKED ISSUE #4639: Improve "disabling l/r panning" accessibility feature discoverability
**Is your feature request related to a problem? Please describe.**
The disable left/right panning feature is hidden behind a chat command, it's hard to discover and the name doesn't make it clear what it does.

**Describe the solution you'd like**
We should improve the feature discoverability exposing it as a preference under Audio/Accessibility so it's easier to find.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/HUBS-896)

### PR DESCRIPTION
**Pull request by keianhzo**
Closes #4639

This PR exposes Left/Right panning as an accessibility setting. Thing to note:
- We no longer use the `audioOutputMode` preference as it has been outlined in the code comments as confusing, now we have a `disableLeftRightPanning` setting and the `audiomode` command updates that pref.
https://github.com/mozilla/hubs/blob/fcf78a30652cb745f90d8e20159ade3031157d2a/src/message-dispatch.js#L179-L181
- This PR removes the initial panning mode pref resetting that was happening before.
https://github.com/mozilla/hubs/blob/fcf78a30652cb745f90d8e20159ade3031157d2a/src/systems/audio-settings-system.js#L8-L17
- Also we no longer log in the console when the mode is changed (we don't do that for any other pref either so it kind of makes sense now that it has graduated to pref)

### CODE REVIEW

File: src/systems/sound-effects-system.js [hunk 1]
```diff
@@ -138,7 +138,7 @@ export class SoundEffectsSystem {
     const audioBuffer = this.sounds.get(sound);
     if (!audioBuffer) return null;
 
-    const disablePositionalAudio = isSafari() || window.APP.store.state.preferences.audioOutputMode === "audio";
+    const disablePositionalAudio = isSafari() || window.APP.store.state.preferences.disableLeftRightPanning === true;
```
Comment (johnshaughnessy) [inline src/systems/sound-effects-system.js:141]: Can `window.APP.store.state.preferences.disableLeftRightPanning === true` be reduced to `window.APP.store.state.preferences.disableLeftRightPanning` here?

File: src/storage/store.js [hunk 2]
```diff
@@ -96,7 +96,6 @@ export const SCHEMA = {
         preferredMic: { type: "string" },
         preferredCamera: { type: "string" },
         muteMicOnEntry: { type: "bool" },
-        audioOutputMode: { type: "string" },
```
Comment (johnshaughnessy) [inline src/storage/store.js:99]: If you remove this key from the store, but it was set in a user's local storage because in their most recent setting they set their `audioOutputMode` pref (via the command), then when the user loads the page, the data in their local storage will cause them to fail the schema validation and the page will fail to load.

Before schema validation runs the first time, we need to delete this entry from users' local storage. Alternatively, we can leave it in the schema definition for a while (with code anywhere that deletes the stored value) with a comment that says, "Deprecated, to be removed once we're confident local storage has been cleaned up."

I think I like the first option better, where before doing schema validation we delete now-deprecated keys if they exist.

File: src/message-dispatch.js [hunk 3]
```diff
@@ -174,20 +174,9 @@ export default class MessageDispatch extends EventTarget {
         break;
       case "audiomode":
         {
-          const shouldEnablePositionalAudio = window.APP.store.state.preferences.audioOutputMode === "audio";
           window.APP.store.update({
-            // TODO: This should probably just be a boolean to disable panner node settings
-            // and even if it's not, "audio" is a weird name for the "audioOutputMode" that means
-            // "stereo" / "not panner".
-            preferences: { audioOutputMode: shouldEnablePositionalAudio ? "panner" : "audio" }
+            preferences: { disableLeftRightPanning: args[0] === "panner" ? false : true }
```
Comment (johnshaughnessy) [inline src/message-dispatch.js:178]: We can remove this chat command

File: src/components/avatar-audio-source.js [hunk 4]
```diff
@@ -100,11 +100,11 @@ AFRAME.registerComponent("avatar-audio-source", {
     APP.dialog.on("stream_updated", this._onStreamUpdated, this);
     this.createAudio();
 
-    let audioOutputModePref = APP.store.state.preferences.audioOutputMode;
+    let disableLeftRightPanningPref = APP.store.state.preferences.disableLeftRightPanning === true;
```
Comment (johnshaughnessy) [inline src/components/avatar-audio-source.js:103]: I think you can drop `=== true`.

**Review by johnshaughnessy — CHANGES_REQUESTED** [review]
My only major concern has to do with removing keys from the `store`. If a user has a key-value pair for `audioOutputMode`, this change will cause the schema validation to fail, which is an error we currently do not catch or recover from.

File: src/systems/sound-effects-system.js (hunk 1 above)
Comment (keianhzo) [reply src/systems/sound-effects-system.js:141]: yep, I think so, we are only using it for the ternary. Updated.

File: src/message-dispatch.js (hunk 3 above)
Comment (keianhzo) [reply src/message-dispatch.js:178]: I guess there is no point on keeping that around for compat. Removed.

File: src/components/avatar-audio-source.js (hunk 4 above)
Comment (keianhzo) [reply src/components/avatar-audio-source.js:103]: In this case (same as in media-video) I think we want to only recreate in case it went from undefined/false <-> true and not if it changed from undefined <-> false

Comment (keianhzo) [reply src/components/avatar-audio-source.js:103]: Thinking about it again now that we don't have the chat command, I don't think there is a case where this goes from `undefined` to `false` so I think is safe to remove that comparison.

File: src/storage/store.js (hunk 2 above)
Comment (keianhzo) [reply src/storage/store.js:99]: I think we switched the preferences store to accept additional properties here:
https://github.com/mozilla/hubs/pull/4275/commits/e865eb78812980297b6ce6fae5236e3b737b0911#r636548037
That's why it doesn't really crash (and also why it doesn't crash even though I forgot to add the actual new `disableLeftRightPanning` pref 😬) I wonder if we should write cleanup code though so old prefs are removed and the store doesn't hold stall prefs.

**Comment by keianhzo** [comment]
@johnshaughnessy I've fixed all the issues and added a final commit to always remove unsupported properties from the store after validating. As we don't want to accept unsupported properties I think this should help keep the store clean.

File: src/storage/store.js (hunk 2 above)
Comment (johnshaughnessy) [reply src/storage/store.js:99]: Oh, great!

File: src/storage/store.js [hunk 5]
```diff
@@ -360,12 +359,13 @@ export default class Store extends EventTarget {
...
+    const { valid, errors } = validator.validate(finalState, SCHEMA);
 
+    // Cleanup unsupported properties
     if (!valid) {
-      // Intentionally not including details about the state or validation result here, since we don't want to leak
-      // sensitive data in the error message.
-      throw new Error(`Write to store failed schema validation.`);
+      errors.forEach(error => {
+        delete error.instance[error.argument];
+      });
```
Comment (johnshaughnessy) [inline src/storage/store.js:368]: It seems like we should at least log these errors. Also, is this change necessary/desirable given the `allowAdditionalProps` change you referred to in an earlier comment.

File: src/update-audio-settings.js [hunk 6]
```diff
@@ -36,7 +36,7 @@ export function getCurrentAudioSettings(el) {
   const audioDebugPanelOverrides = APP.audioDebugPanelOverrides.get(sourceType);
   const zoneSettings = APP.zoneOverrides.get(el);
   const preferencesOverrides =
-    APP.store.state.preferences.audioOutputMode === "audio" ? { audioType: AudioType.Stereo } : {};
+    APP.store.state.preferences.disableLeftRightPanning === true ? { audioType: AudioType.Stereo } : {};
```
Comment (johnshaughnessy) [inline src/update-audio-settings.js:39]: We can lose `=== true`

**Review by johnshaughnessy — APPROVED** [review]

File: src/storage/store.js (hunk 5 above)
Comment (keianhzo) [reply src/storage/store.js:369]: First comment here https://github.com/mozilla/hubs/pull/4275/files#r636548037 talked about cleaning up the store to avoid stall prefs to stick around forever. I thought it was a good time for adding that.
